python/dgl/data/gindt.py

```diff
@@ -0,0 +1,270 @@
+"""Dataset for Graph Isomorphism Network(GIN)
+(chen jun): Used for compacted graph kernel dataset in GIN
+
+Data sets include:
+
+MUTAG, COLLAB, IMDBBINARY, IMDBMULTI, NCI1, PROTEINS, PTC, REDDITBINARY, REDDITMULTI5K
+https://github.com/weihua916/powerful-gnns/blob/master/dataset.zip
+"""
+
+import os
+import logging
+import torch
+import numpy as np
+
+from .utils import download, extract_archive, get_download_dir, _get_dgl_url
+from ..graph import DGLGraph
+
+
+_url = 'https://raw.githubusercontent.com/weihua916/powerful-gnns/master/dataset.zip'
+
+
+class GINDataset(object):
+    """Datasets for Graph Isomorphism Network (GIN)
+    Adapted for the paper 'how powerful are graph neural networks?'
+
+    The dataset contains the compact format of popular graph kernel datasets,
+    which included:
+    MUTAG, COLLAB, IMDBBINARY, IMDBMULTI, NCI1, PROTEINS, PTC, REDDITBINARY, REDDITMULTI5K
```

Comment:
Maybe add a pointer to `TUDataset` so user will know how to access other graph datasets?

Reply:
> Maybe add a pointer to `TUDataset` so user will know how to access other graph datasets?

good idea

Reply:
there is an error when checked cpp code but I never touch cpp part, see detail https://discuss.dgl.ai/t/category-build-errors-found-when-checking-by-task-lint/186

Reply:
> Maybe add a pointer to `TUDataset` so user will know how to access other graph datasets?
"
This datset class processes all data sets listed above. For more graph kernel datasets, 
    see :class:`TUDataset`  
"
done